Automattic/jetpack
# Mobile Theme: remove feature

Fixes #13385

Closes #6616 
Closes #5815
Closes #6617
Closes #10685
Closes #2061

#### Changes proposed in this Pull Request:

It is time.

We originally created the mobile theme feature as a fall-back when the regular theme did not include a mobile view. Most themes include a mobile view by default now, so the feature is no longer necessary.

This PR is the last step in deprecating the Mobile Theme feature. In #14101 (as well as messages in the dashboard and emails), we started warning folks that the feature was going away. In this PR, we remove the feature entirely:

- We deactivate the module if it still active.
- We deactivate the module on plugin update.
- We delete options created by the module to clean things up.
- We remove the UI from the Jetpack Settings dashboard.
- We output notices to folks calling functions that were used by the module, or calling the module file directly.
- We stop synchronizing options with WordPress.com.
- We output notices to all folks who may still use filters / hooks to customize the mobile theme.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* It removes an existing feature. Internal reference: p1HpG7-85q-p2

#### Testing instructions:

* Start from a site that does not run this branch.
* Run `wp jetpack module activate minileven` to activate the Mobile Theme. 
* Switch to this branch.
* The Mobile Theme should not be active when you visit your site on a mobile device.
* You should see no PHP errors.

#### Proposed changelog entry for your changes:

* Mobile Theme: sunset feature. We originally created the mobile theme feature as a fall-back when the regular theme did not include a mobile view. Most themes include a mobile view by default now, so the feature is no longer necessary.
